pallets/pallet-bonded-coins/src/lib.rs

```diff
@@ -169,66 +176,178 @@ pub mod pallet {
 	impl<T: Config> Pallet<T> {
 		#[pallet::call_index(0)]
 		#[pallet::weight(Weight::from_parts(10_000, 0) + T::DbWeight::get().writes(1))]
-		pub fn create_pool(_origin: OriginFor<T>) -> DispatchResult {
-			todo!()
+		pub fn create_pool(
+			origin: OriginFor<T>,
+			curve: CurveInput<CurveParameterInputOf<T>>,
+			currencies: BoundedVec<TokenMetaOf<T>, T::MaxCurrencies>,
+			denomination: u8,
+			pool_manager: AccountIdOf<T>,
+			transferable: bool,
+		) -> DispatchResult {
+			let who = T::PoolCreateOrigin::ensure_origin(origin)?;
+
+			let currency_length = currencies.len();
+
+			let checked_curve = curve.try_into().map_err(|_| Error::<T>::InvalidInput)?;
+
+			let current_asset_id = NextAssetId::<T>::get();
```

Comment:
@ntn-x2 suggested to have users supply the asset ids to avoid the situation that we reach the numerical bounds of the asset id type and cannot safely wrap around because some of the ids we would be trying to use then are still in use. Did we decide this is a non-issue?

On a related note, the assets pallet now has the exact same functionality - not sure it exposes it via any public trait though.

Reply:
Maybe this can be a callback / hook instead, so we can implement this in the runtime to rely on the asset pallet's id increment solution. This would also avoid the pretty bad situation where the `NextAssetId` gets registered without this pallet knowing (e.g., via the asset pallet's extrinsics), which would break pool creation until governance steps in and force-increments this.

Reply:
If the asset ID in here is strictly depending on some conditions of the assets pallet, then I would recommend making it a injectable hook for exactly the reason where a new asset class is created in that pallet and this pallet is not aware of it.

Reply:
> Did we decide this is a non-issue?

Yes, we did. This would only happen if there were already over 4 million bonded currencies. If that ever becomes a problem, it would mean we’re very successful,

> This would also avoid the pretty bad situation where the NextAssetId gets registered without this pallet knowing (e.g., via the asset pallet's extrinsics)

This shouldn’t be an issue. In the runtime, only root can create new assets. Of course, if there’s a public referendum to create a new asset in the pallet, it should be blocked. 


> If the asset ID in here is strictly depending on some conditions of the assets pallet, then I would recommend making it a injectable hook 

I’m worried that adding a hook could cause issues later. Right now, there’s no `NextAssetId` struct in the fungibles pallet, but this feature is coming in version 14. We need this storage map temporarily. If we remove the storage value completely and allow users to define any asset ID not in use, we wouldn’t be able to migrate to this feature later. We’d also lose control over which assets exist.

Reply:
Yes, that would be possible and also my preferred way. I would keep the `NexAssetId` storage value, until we hit a version, where the feature is implemented, and we can migrate over and use a hook.